src/Denudation/Abstract.jl

```diff
@@ -24,7 +24,13 @@ function denudation(input)
             end
 
             if water_depth[idx] >= 0
-                (denudation_mass[idx]) = denudation(input.box, input.denudation, water_depth[idx], slope[idx], input.facies[f])
+                
+                if denudation(input.box, input.denudation, water_depth[idx], slope[idx], input.facies[f]) !== nothing
```

Comment:
You're calling this function twice here. In many cases the compiler will optimize this away, but since it is quite a long call, it is more hygienic to write it differently.